src/IssueViz.Security/Taint/SonarQubeIssueToIssueVisualizationConverter.cs

```diff
@@ -0,0 +1,127 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Taint
+{
+    interface ISonarQubeIssueToIssueVisualizationConverter
+    {
+        IAnalysisIssueVisualization Convert(SonarQubeIssue sonarQubeIssue);
+    }
+
+    [Export(typeof(ISonarQubeIssueToIssueVisualizationConverter))]
+    internal class SonarQubeIssueToIssueVisualizationConverter : ISonarQubeIssueToIssueVisualizationConverter
+    {
+        private readonly IAnalysisIssueVisualizationConverter issueVisualizationConverter;
+        private readonly IAbsoluteFilePathLocator absoluteFilePathLocator;
+
+        [ImportingConstructor]
+        public SonarQubeIssueToIssueVisualizationConverter(IAnalysisIssueVisualizationConverter issueVisualizationConverter, IAbsoluteFilePathLocator absoluteFilePathLocator)
+        {
+            this.issueVisualizationConverter = issueVisualizationConverter;
+            this.absoluteFilePathLocator = absoluteFilePathLocator;
+        }
+
+        public IAnalysisIssueVisualization Convert(SonarQubeIssue sonarQubeIssue)
+        {
+            var analysisIssue = ConvertToAnalysisIssue(sonarQubeIssue);
+            var issueViz = issueVisualizationConverter.Convert(analysisIssue);
+
+            return issueViz;
+        }
+
+        private IAnalysisIssueBase ConvertToAnalysisIssue(SonarQubeIssue sonarQubeIssue)
+        {
+            if (sonarQubeIssue.TextRange == null)
```

Comment:
I don't think it's possible to have a vulnerability without a span or to have a location without a span, but I'll check tomorrow with Alex/Marco.